app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -100,7 +100,11 @@ def parse_cluster(cluster)
       }
       set_relationship_on_hosts(new_result, nodes)
 
-      return uid, new_result
+      if @data[:hosts].select { |host| host[:ems_cluster] == new_result }.empty?
+        return nil
+      else
+        return uid, new_result
+      end
```

Comment:
I would prefer the `any?` syntax here:

```ruby
if @data[:hosts].any? { |host| host[:ems_cluster] == new_result }
  return uid, new_result 
else
  return nil
end
```

The `return nil` is implied, so you could probably even one-line this like:

```ruby
return uid, new_result if @data[:hosts].any? { |host| host[:ems_cluster] == new_result }
```